### DISCUSSION TITLE
Soft key shortcuts on tooltips

### DISCUSSION THREAD

**Discussion by sheepo99** [OpenTTD/OpenTTD#11740] [Features]
Hello,

Previously we discussed the possibility of implementing soft key shortcuts to be displayed on tooltips, as part of improving user accessibility.

## Implementation

The idea is that every tooltip displays its keyboard shortcut that is sourced from the configurable keyboard shortcuts file automatically via a function string (e.g. "Build railway signals (*function string here*)". Keys themselves should also have translatable string names and the ability to automatically convert these to small font after a certain number of characters (for key names such as Spacebar and Return).

**Comment by 2TallTyler** [comment]
Relevant PR: #9732.
